packages/table/src/CheckboxCell.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Checkbox from '@leafygreen-ui/checkbox';
+import { css } from '@leafygreen-ui/emotion';
+import { useTableContext, Types } from './TableContext';
+
+const flexStyles = css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+interface CheckboxCellProps extends React.ComponentPropsWithoutRef<'td'> {
+  checked?: boolean;
+  disabled?: boolean;
+  index?: number;
+}
+
+function CheckboxCell({
+  children,
+  className,
+  disabled,
+  index,
+}: CheckboxCellProps) {
+  const {
+    state: { headerCheckState },
+    dispatch,
+  } = useTableContext();
+  const [checked, setChecked] = React.useState(headerCheckState);
```

Comment:
Is this just to make things look faster?

We shouldn't do this if we don't have to and just fix the core underlying performance issues.

Reply:
We know what to toggle it to when we actually dispatch the event, right before we set the new value we pull the old one and invert it.

Reply:
Removed!